common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -875,6 +875,14 @@ def user_details_force_sync(auth_entry, strategy, details, user=None, *args, **k
                 changed[provider_field] = current_value
                 setattr(model, field, provider_value)
 
+        # Generate fullname only for IES IDP.
+        if (not details.get('fullname') and details.get('first_name') and details.get('last_name') and
```

Comment:
Let's put this code under `try except` logging if an error happens. Something like 'u'[THIRD_PARTY_AUTH] Couldn't generate full name for user: %''

Reply:
Done.